### ISSUE TITLE
something went wrong with Listing 16-2. stack_unwind.asm

### ISSUE THREAD

**Issue by kneoyskw** [Apress/low-level-programming#44]
I built an executable file with the following Makefile in` listings/chap16/stack_unwind`. When executed, the program printed **14**  lines (not 10) and caused Segmentation fault. I think something is wrong with the loop termination logic in` stack_unwind.asm`.  I don't know how `rbx` register value become zero. Any suggestion?

```
all: main

main: stack_unwind.o main.o 
	gcc -o main main.o stack_unwind.o

stack_unwind.o: stack_unwind.asm
	nasm -felf64 stack_unwind.asm -o stack_unwind.o

main.o: stack_unwind.c
	gcc -c -O0 stack_unwind.c -o main.o
```

**Comment by kneoyskw** [comment]
I figured out another way to stop looping and made a pull request of my solution. I believe that a better solution could be found but I just wanted to show what is the problem.

**Comment by sayon** [comment]
Fixed in #45. Thank you again!
